File: leetcode-solutions/greedy/solution-2444/solution.go
package solution2444

import "math"

func countSubarrays(nums []int, minK int, maxK int) (res int64) {
    p, q, s := 1000000, 1000000, 1000000
    for i, v := range nums {
        if v > maxK || v < minK {
            p, q, s = 1000000, 1000000, 1000000
        } else {
            if s == 1000000 {
                s = i
            }
            if v == minK {
                p = i
            }
            if v == maxK {
                q = i
            }
            if p != 1000000 && q != 1000000 {
                res += int64(math.Min(float64(p), float64(q))) - int64(s) + 1
            }
        }
    }
    return
}

File: leetcode-solutions/greedy/solution-2444/solution.js
/**
 * @param {number[]} nums
 * @param {number} minK
 * @param {number} maxK
 * @return {number}
 */
var countSubarrays = function(nums, minK, maxK) {
  let p, q, s;
  let res = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] > maxK || nums[i] < minK) {
      p = q = s = undefined;
    } else {
      if (s === undefined) s = i;
      if (nums[i] === minK) p = i;
      if (nums[i] === maxK) q = i;
      if (p !== undefined && q !== undefined) {
        res += Math.min(p, q) - s + 1;
      }
    }
  }
  return res;
};
